frontend/src/routes/Provider/components/DocsMenu.tsx

```diff
@@ -107,11 +117,23 @@ export function ProviderDocsMenu() {
   const { data } = useSuspenseQuery(
     getProviderVersionDataQuery(namespace, provider, version),
   );
-
+  const [searchFilter, setSearchFilter] = useState("");
```

Comment:
Let's use `useDeferredValue` together with the state above to let React know it can lag behind with re-rendering of the sidebar while user finishes the filter query.